pkg/controller/tf/controller.go

```diff
@@ -199,11 +199,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (res
 			return reconcile.Result{}, err
 		}
 
-		// add finalizers for deletion defender
 		if resource.GetDeletionTimestamp().IsZero() {
+			// add finalizers for deletion defender
 			if err := r.EnsureFinalizers(ctx, resource.Original, &resource.Resource, k8s.ControllerFinalizerName, k8s.DeletionDefenderFinalizerName); err != nil {
 				return reconcile.Result{}, err
 			}
+
+			if err := r.HandlePaused(ctx, resource.Original); err != nil {
```

Comment:
As you put HandlePaused() in the if branch to check whether this is not for deletion - how will deletion work when it is paused? Taking a brief look at the code, iiuc, it just logs `Skipping actuation ...` and continues the re-reconciliation?

This is orthogonal to this PR though I wondered if I've missed anything about the deletion story.

Reply:
This is a good point -- if we bill pause as "reconciliation of k8s resources still happens BUT not actuation" then there's more work needed here to implement that promise.

Reply:
( deleted previous comment as I had misunderstood the situation here )

sorry @maqiuyujoyce I may be missing something obvious but what is the concern for deletion when the resource is paused?